taigapy/client_v3.py

```diff
@@ -312,7 +314,86 @@ def _get(self, datafile_id: str) -> pd.DataFrame:
 
         return result
 
-    def download_to_cache(self, datafile_id: str, requested_format: Union[LocalFormat, str]) -> str:
+    def upload_to_gcs(
+        self,
+        data_file_taiga_id: str,
+        requested_format: LocalFormat,
+        dest_gcs_path_for_file: str,
+    ) -> bool:
+        """Upload a Taiga datafile to a specified location in Google Cloud Storage.
+
+        The service account taiga-892@cds-logging.iam.gserviceaccount.com must have
+        storage.buckets.create access for this request.
+
+        Arguments:
+            `data_file_taiga_id` -- Taiga ID in the form dataset_permaname.dataset_version/datafile_name
+            `requested_format` -- The format of the file you want to upload
+            `dest_gcs_path_for_file` -- Google Storage path to upload to, in the form 'gs://bucket:file_obj_name'
+
+        Returns:
+            bool -- Whether the file was successfully uploaded
+        """
+
+        def parse_gcs_path(gcs_path: str) -> Tuple[str, str]:
+            gcs_path_error = ValueError(
+                "Invalid GCS path. '{}' is not in the form 'gs://bucket_name/object_name'".format(
+                    gcs_path
+                )
+            )
+            if not gcs_path.startswith("gs://"):
+                raise gcs_path_error
+            # remove prefix
+            gcs_path = gcs_path.replace("gs://", "")
```

Comment:
This can be problematic as it'll corrupt a path if it contained "gs://" in the middle of the string. Very unlikely in this specific circumstance. I prefer regex's for string parsing like this because they are more explict as how they break the string up.

For example, you could do:


```
m = re.match("gs://([^/]+)/(.*)$", gs_path)
if not m:
  raise ValueError(
                "Invalid GCS path. '{}' is not in the form 'gs://bucket_name/object_name'".format(
                    gcs_path
                )
            )
bucket_name, file_object_name = m.groups()

Reply:
It's more robust ... and less typing. :)

Reply:
SOunds goOd!